### ISSUE TITLE
Reproduce the result in paper

### ISSUE THREAD

**Issue by Sanqiang** [facebookresearch/access#28]
I git clone the `submission` branch.
Installed all the libraries in `requirement.txt`
Then I ran `scripts/train.py` without changing anything.
It seems like the validation SARI keeps around 39+, cannot reach 41+

Could I know what I am missing?

**Comment by louismartin** [comment]
Hi Sanqiang, thanks for the issue!

From what I recall, the results had a lot of variance unfortunately, so we ran the trainings a few dozen times before selecting the best model with SARI on the validation set.

Hope that helps, tell me if you have more questions!

**Comment by Sanqiang** [comment]
Thanks, 
Could I confirm a question: acutually the SARI I mean it is the validation SARI generated by the end of each epoch (calcualted by easse)? It seems like it keeps around 39+.
Could I confirm the how the validation compute? It is based on all the control ratio equals to 1?

**Comment by louismartin** [comment]
Oh then that's normal I think, it's because during validation we don't optimize for the ACCESS control values, I think we keep them all fixed at 0.8 ratios.
When the training is finished, it's at this time that we run the zero-order optimization (i.e. just an fancy grid search), to find the best control values (e.g. length=0.8, levenshtein=0.65, lexical_complexity=0.6 ...).
So yes the validation SARI will always be somewhat lower!

**Comment by Sanqiang** [comment]
I see.
The validation SARI during the training is based on all the ratios at 0.8?
Did you update those ratios during training (for validation SARI).

**Comment by louismartin** [comment]
Yes exactly, we keep them frozen to 0.8 for validation SARI and we never update them.
Then once the full training is finished (e.g. because of early stopping on this validation SARI), we run the grid search to find the best ratios on the validation set and use these ratios to report the performance on the test set.

**Comment by louismartin** [comment]
Hi @Sanqiang , can I close the issue now ?

**Comment by louismartin** [comment]
Closing for now, feel free to comment or open again if you are still having issues.
